app/src/ApiClient.ts

```diff
@@ -187,49 +192,40 @@ export class ApiClient {
     });
   }
 
-  private async populateClient(): Promise<AxiosInstance> {
-    if (!this.client) {
-      this.client = this.buildClient();
-    }
-    return this.client;
-  }
-
   async loginUrl(): Promise<string> {
-    return (await this.populateClient()).getUri({ url: "/login" });
-  }
-
-  async redirectToLogin(): Promise<null> {
-    const loginUrl = await this.loginUrl();
-    window.location.href = loginUrl;
-    return null;
+    return (await this.#client).getUri({ url: "/login" });
   }
 
   async logoutUrl(): Promise<string> {
-    return (await this.populateClient()).getUri({ url: "/logout" });
+    return (await this.#client).getUri({ url: "/logout" });
   }
 
   async getCurrentUser(): Promise<User> {
     const res = await this.get("/api/users/me");
     return res.data as User;
   }
 
+  async currentUser(): Promise<User> {
+    return (this.#currentUser ??= this.getCurrentUser());
```

Comment:
this caching line removes the double fetch when loading `/` and being "redirected" within the router to `/accounts`, both of which call the layout's loader because it has not finished yet. apparently react-router doesn't track in-flight loaders in order to prevent calling them twice, so we track that in the api client. this is safe to cache for the lifetime of the app because current user will never change without a document reload